File: it355/it355-pz01-nikola_tasic_3698/src/main/java/com/example/backend/entity/Comment.java
package com.example.backend.entity;

import lombok.Data;
import lombok.EqualsAndHashCode;

import javax.persistence.*;

/**
 * User comment on a post
 */
@Data
@Entity
@Table(name = "comment")
@EqualsAndHashCode(callSuper = false, onlyExplicitlyIncluded = true)
public class Comment extends Auditable {
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@EqualsAndHashCode.Include
	@Column(name = "comment_id")
	private Integer id;
	@JoinColumn(name = "user_fk", referencedColumnName = "user_id")
	@ManyToOne
	private User user;
	@JoinColumn(name = "post_fk", referencedColumnName = "post_id")
	@ManyToOne
	private Post post;
	@Column(name = "body")
	private String body;
}

File: cs322/cs322-pz-nikola_tasic_3698/blog-backend/Data/Entity/Comment.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace blog_backend.Data;

[Table("comment")]
public class Comment : Auditable {
	[Key, Column("comment_id")]
	public int Id { get; set; }

	[Column("user_fk")]
	[ForeignKey("User")]
	public int? UserFk { get; set; }

	[ForeignKey("UserFk")]
	public virtual User? User { get; set; }

	[Column("post_fk")]
	[ForeignKey("Post")]
	public int? PostFk { get; set; }

	[ForeignKey("PostFk")]
	public Post Post { get; set; }

	[Column("body")]
	public string Body { get; set; }

}